File: apps/ios/Pebbles/Features/Path/Models/ComposePebbleResponse.swift
import Foundation

/// Decodable wrapper for the `compose-pebble` edge function response.
///
/// Fields are optional because the edge function may return a soft-success
/// 5xx body with only `pebble_id` set when the insert succeeded but the
/// compose step failed. The iOS client advances to the detail sheet in that
/// case; the sheet renders text-only when `renderSvg` is nil.
struct ComposePebbleResponse: Decodable {
    let pebbleId: UUID
    let renderSvg: String?
    let renderVersion: String?

    enum CodingKeys: String, CodingKey {
        case pebbleId = "pebble_id"
        case renderSvg = "render_svg"
        case renderVersion = "render_version"
    }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/models/ComposePebbleResponse.kt
package app.pbbls.android.features.path.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Decoded body of the `compose-pebble` / `compose-pebble-update` edge-function
 * response — mirrors iOS `ComposePebbleResponse.swift`.
 *
 * All fields except [pebbleId] are nullable because a soft-success 5xx body
 * carries only `pebble_id` when the insert succeeded but the compose step
 * failed; the client still advances to the detail reveal and renders text-only
 * when [renderSvg] is null.
 */
@Serializable
data class ComposePebbleResponse(
    @SerialName("pebble_id")
    val pebbleId: String,
    @SerialName("render_svg")
    val renderSvg: String? = null,
    @SerialName("render_version")
    val renderVersion: String? = null,
    @SerialName("karma_delta")
    val karmaDelta: Int? = null,
)
